docs/reference/api/platform/Error.md

```diff
@@ -1,14 +1,123 @@
-## Error
+## Mbed OS Error Status Definitions and Error Handling
+Mbed OS provides error code and error status definitions and error handling APIs for Error construction, reporting and retrieving previously reported errors. It also provides functions and macros to generate and define new error status values, extract information from error status values and to generate fatal system errors which terminates the application. Any software layer(Applications, Drivers, HAL, Protocol stacks etc) can use these error handling APIs. The error functions also facilitates emitting an error message through STDOUT. `mbed_error.h` declares the error functions provided by Mbed OS.
 
-Mbed OS provides an error function to output messages to `STDIO` at runtime when the system encounters a fatal error and the Application calling the error function will be terminated. Note that the error function outputs the error message in [debug and develop builds](/docs/development/tools/build-profiles.html) only. In release builds, the error function does not generate any `STDIO` output, but the application is still terminated. `mbed_error.h` declares the error function.
+Conceptually, Error Handling is a platform service implemented by platform layer of Mbed OS and provides the following:
```

Comment:
Super small nit: "...is a platform service implemented by *the* platform layer..."